data_processors/pipeline/lambdas/somalier_extract.py

```diff
@@ -90,17 +91,23 @@ def handler(event, context) -> dict:
     timestamp = int(datetime.now().replace(microsecond=0).timestamp())
     step_function_instance_name = "__".join([
         "somalier_extract",
-        urlparse(gds_path).path.lstrip("/").replace("/", "_").rstrip(".bam")[-40:],
+        urlparse(index).path.lstrip("/").replace("/", "_").rstrip(".bam")[-40:],
```

Comment:
Would a function get_somalier_step_function_instance_name be a better implementation here? Takes two inputs, index and function start name, might make it easier to document the expected input (index) and what we can expect as an output step_function_instance_name

Reply:
Good point. Let me refactor.

Reply:
As suggested; refactored with https://github.com/umccr/data-portal-apis/commit/9ec36a2d507dd1318658268424e101fe7cc7e80e